packages/js/src/server/integrations/uncaught_exception.ts

```diff
@@ -34,32 +34,32 @@ export default function (): Types.Plugin {
 
       removeAwsLambdaListener()
 
-      const hasOtherListeners = hasUncaughtExceptionListeners();
+
       process.on('uncaughtException', function honeybadgerUncaughtExceptionListener(uncaughtError) {
         if (!client.config.enableUncaught) {
           client.config.afterUncaught(uncaughtError)
-          if (!hasOtherListeners) {
+          if (!hasOtherUncaughtExceptionListeners()) {
             fatallyLogAndExit(uncaughtError)
           }
           return
         }
 
         // report only the first error - prevent reporting recursive errors
-        if (count < 1) {
+        if (!reportedOnce) {
           isReporting = true
           client.notify(uncaughtError, {
             afterNotify: (_err, _notice) => {
               isReporting = false
-              count++
+              reportedOnce = true
```

Comment:
Good points!

The early return is a bit tricky though, I tried it and it's not as clean as you show it above:
- we need to handle the case where `handlerAlreadyCalled` is `false` but we don't want to report to HB because `isReporting` is `true`

But I may have a solution with multiple early returns. Incoming commit!